dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/FileAssetTemplateUtil.java

```diff
@@ -0,0 +1,497 @@
+ */
+public class FileAssetTemplateUtil {
+
+    private static final String TITLE                = "title";
+    private static final String FRIENDLY_NAME = "friendly_name";
+    private static final String IMAGE                = "image";
+    private static final String THEME                = "theme";
+    private static final String HOST_INDICATOR       = "//";
+
+    private static String [] DEFAULT_META_DATA_NAMES_ARRAY = new String[] { TITLE, FRIENDLY_NAME,THEME,IMAGE };
+
+    public static final String LAYOUT             = "layout.json";
+    public static final String BODY               = "body.vtl";
+    public static final String TEMPLATE_META_INFO = "properties.vtl";
+
+    private static class SingletonHolder {
+        private static final FileAssetTemplateUtil INSTANCE = new FileAssetTemplateUtil();
+    }
+    /**
+     * Get the instance.
+     * @return FileAssetTemplateUtil
+     */
+    public static FileAssetTemplateUtil getInstance() {
+
+        return FileAssetTemplateUtil.SingletonHolder.INSTANCE;
+    } // getInstance.
+
+    /**
+     * Determines if the template id is a path or uuid, if not any match will return UNKNOWN.
+     * @param templateIdOrPath String path or uddi
+     * @return Source (File if it is a fs template, DB if it is a db template, otherwise UNKNOWN)
+     */
+    public Source getTemplateSourceFromTemplateIdOrPath(final String templateIdOrPath) {
+
+        Source source = Source.UNKNOWN;
+
+        if (this.isFolderAssetTemplateId(templateIdOrPath)) {
+            source = Source.FILE;
+        } else if (this.isDataBaseTemplateId(templateIdOrPath)) {
+            source = Source.DB;
+        }
+
+        return source;
+    }
+
+    public boolean isDataBaseTemplateId(final String templateIdentifier) {
+
+        boolean isIdentifier = false;
+        isIdentifier |= UUIDUtil.isUUID(templateIdentifier);
+
+        if (!isIdentifier) {
+            try {
+                APILocator.getShortyAPI().validShorty(templateIdentifier);
+                isIdentifier = true;
+            } catch (Exception e) {
+                isIdentifier = false;
+            }
+        }
+
+        return isIdentifier;
+    }
+
+    public Host getHost (final String path) throws DotSecurityException, DotDataException {
+        return this.getHostFromHostname(this.getHostName(path));
+    }
+
+    public Host getHostFromHostname (final String hostname) throws DotSecurityException, DotDataException {
+        return null == hostname?
+                APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false) :
+                APILocator.getHostAPI().resolveHostName(hostname, APILocator.systemUser(), false);
+    }
+
+    final List<String> pageModePrefixList = Stream.of(PageMode.values())
+            .map(pageMode -> String.format("/%s/", pageMode.name()))
+            .collect(Collectors.toList());
+
+
+    //demo.dotcms.com/application/templates/test/
+    public String getHostName(final String path) {
```

Comment:
I think this logic is kinda similar to the container, probably would be a good idea to unified the logic in just one class and pass more parameters (for container or templates)
In that case, if we found an error probably can fix just once